pkg/codec/by_item_type_modifier_test.go

```diff
@@ -0,0 +1,50 @@
+package codec_test
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+func TestByTypeModifier(t *testing.T) {
+	anyitemType := "anything"
```

Comment:
nit/ prefer const if possible?
```suggestion
	const anyitemType = "anything"
```